src/usr/share/netcheck/netcheck-daemon.py

```diff
@@ -17,86 +17,62 @@
 
 import confighelper
 import ConfigParser
+import daemon
 import logging
 import netcheck
 import os
+# TODO: Remove try/except when we drop support for Ubuntu 14.04 LTS.
+try:
+    from lockfile import pidlockfile
+
+except ImportError:
+    from daemon import pidlockfile
+
 import signal
 import sys
 import traceback
 
 PID_FILE = '/run/netcheck.pid'
 
-
-# TODO: Use standard daemonize module.
-def daemonize():
-    # Fork the first time to make init our parent.
-    try:
-        pid = os.fork()
-        if pid > 0:
-            sys.exit(0)
-    except OSError as e:
-        logger.critical('Failed to make parent process init: %d (%s)' %
-                        (e.errno, e.strerror))
-        sys.exit(1)
-
-    # TODO: Lock this down.
-    os.chdir('/')  # Change the working directory
-    os.setsid()  # Create a new process session.
-    os.umask(0)
-
-    # Fork the second time to make sure the process is not a session leader.
-    #   This apparently prevents us from taking control of a TTY.
-    try:
-        pid = os.fork()
-        if pid > 0:
-            sys.exit(0)
-    except OSError as e:
-        logger.critical('Failed to give up session leadership: %d (%s)' %
-                        (e.errno, e.strerror))
-        sys.exit(1)
-
-    # Redirect standard file descriptors
-    sys.stdout.flush()
-    sys.stderr.flush()
-    devnull = os.open(os.devnull, os.O_RDWR)
-    os.dup2(devnull, sys.stdin.fileno())
-    os.dup2(devnull, sys.stdout.fileno())
-    os.dup2(devnull, sys.stderr.fileno())
-    os.close(devnull)
-
-    pid = str(os.getpid())
-    pidFile = file(PID_FILE, 'w')
-    pidFile.write('%s\n' % pid)
-    pidFile.close()
-
 # Verify config file here.
 
 config_file = ConfigParser.SafeConfigParser()
 config_file.read('/etc/netcheck/netcheck.conf')
 
 # Get logging options first.
 config_helper = confighelper.ConfigHelper()
-log_file = config_helper.verify_string_exists_prelogging(config_file, 'log_file')
-log_level = config_helper.verify_string_exists_prelogging(config_file, 'log_level')
+log_file = config_helper.verify_string_exists(config_file, 'log_file')
+log_level = config_helper.verify_string_exists(config_file, 'log_level')
 
 # Configure the logger.
 config_helper.configure_logger(log_file, log_level)
 logger = logging.getLogger()
 
+def strip_a_list_of_strings(list_of_strings):
```

Comment:
Strangely, I think calling this method a sorter name will make it more clear. How about strip_strings or strip_string_list?

Reply:
Also, I'd name the parameter delimited_string.

Reply:
The parameter is actually a list object of strings, not a string itself.

Reply:
In that case, I'd call the parameter "strings". Let the plural hint that it is a container.